### PR TITLE
Chatwheel Sync

### PR DESCRIPTION
**Pull request by e9k**
Dota2 SIde

### CODE REVIEW

File: game/scripts/vscripts/common/webapi.lua [hunk 1]
```diff
@@ -171,7 +172,23 @@ function WebApi:AfterMatch(winnerTeam)
 		WebApi:Send("match/after", requestBody)
 	end
 end
-
+function WebApi:PatreonPlayerUpdate(updates)
+	local requestBody = {
+		customGame = WebApi.customGame,
+		matchId = isTesting and RandomInt(1, 10000000) or tonumber(tostring(GameRules:GetMatchID())),
+		mapName = GetMapName(),
```
Comment (ark120202) [inline game/scripts/vscripts/common/webapi.lua]: These are unused

File: game/scripts/vscripts/common/patreons.lua [hunk 2]
```diff
@@ -36,9 +36,16 @@ function Patreons:GetPlayerEmblemColor(playerId)
...
 
+pendingPlayerUpdates = {}
 function Patreons:SetPlayerSettings(playerId, settings)
+
 	Patreons.playerSettings[playerId] = settings
 	CustomNetTables:SetTableValue("game_state", "patreon_bonuses", Patreons.playerSettings)
+	pendingPlayerUpdates[playerId] = settings
+	GameRules:GetGameModeEntity():SetThink(function()
+		WebApi:PatreonPlayerUpdate(pendingPlayerUpdates)
+		q = {}
```
Comment (ark120202) [inline game/scripts/vscripts/common/patreons.lua]:
```suggestion
		pendingPlayerUpdates = {}
```

File: game/scripts/vscripts/common/patreons.lua [hunk 3]
```diff
@@ -82,6 +89,8 @@ RegisterCustomEventListener("patreon_toggle_emblem", function(args)
 end)
 
 -- TODO: It's not really related to patreon, it'd be better to extract player options to a different module
+
```
Comment (ark120202) [inline game/scripts/vscripts/common/patreons.lua]: Unrelated spacing changes should be reverted

File: game/scripts/vscripts/common/patreons.lua [hunk 4]
```diff
@@ -36,9 +36,16 @@ function Patreons:GetPlayerEmblemColor(playerId)
 	return colorNames[Patreons:GetPlayerSettings(playerId).emblemColor]
 end
 
+pendingPlayerUpdates = {}
```
Comment (ark120202) [inline game/scripts/vscripts/common/patreons.lua]:
```suggestion
Patreons.pendingPlayerUpdates = {}
```

File: game/scripts/vscripts/common/webapi.lua [hunk 5]
```diff
@@ -171,7 +172,23 @@ function WebApi:AfterMatch(winnerTeam)
...
+		players = {}
+	}
+	for key, value in pairs(updates) do
```
Comment (ark120202) [inline game/scripts/vscripts/common/webapi.lua]: Use more descriptive variable names, i.e.
```suggestion
	for playerId, patreonUpdate in pairs(updates) do
```

File: game/scripts/vscripts/common/patreons.lua [hunk 6]
```diff
@@ -37,8 +38,14 @@ function Patreons:GetPlayerEmblemColor(playerId)
 end
 
 function Patreons:SetPlayerSettings(playerId, settings)
+
 	Patreons.playerSettings[playerId] = settings
 	CustomNetTables:SetTableValue("game_state", "patreon_bonuses", Patreons.playerSettings)
+	Patreons.pendingPlayerUpdates[playerId] = settings
+	GameRules:GetGameModeEntity():SetThink(function()
```
Comment (ark120202) [inline game/scripts/vscripts/common/patreons.lua:45]: `Patreons:SetPlayerSettings` is also called from `WebApi:BeforeMatch`. It should have an optional parameter which makes it ignore the update, and it should be set to `true` in that code path.

**Comment by darklordabc** [comment]
@ark120202 this is up to you if you want to merge
